### PR TITLE
Support [include] in options files used by supervisorctl

### PR DESCRIPTION
**Pull request by farialima**
This is a fix for **numerous** issues:  
copied & modified config 'include' code from ServerOptions to ClientOpti... #277
supervisorctl ignores included configuration files #962
ctlplugin ignored in include files #993
supervisorctl doesn't load included config files, misleading and undocumented #1007 
does supervisorctl actually require the configuration file? #498  (in part)
unix:///var/run/supervisor.sock no such file #480 (very probably part of the confusion)

Maybe that would be worth merging ?

there was already a pull request in #277 but it wasn't up-to-date anymore... this one should be OK, please review and tell me if you would like anything changed.

### CODE REVIEW

**Comment by farialima** [comment]
Hi ! What would be needed to get this merged ? Would you need me to add some tests ? Please tell me...

**Comment by mnaberez** [comment]
Note for maintainers: also credit @jsipprell (#277) in changelog

File: supervisor/options.py [hunk 1]
```diff
@@ -393,6 +393,38 @@ def import_spec(self, spec):
...
+            files = files.split()
+            if hasattr(fp, 'name'):
+                base = os.path.dirname(os.path.abspath(fp.name))
+            else:
+                base = '.'
+            for pattern in files:
+                pattern = os.path.join(base, pattern)
+                filenames = glob.glob(pattern)
+                if not filenames:
+                    self.parse_warnings.append(
```
Comment (mnaberez) [inline supervisor/options.py]: `self.parse_warnings` and `self.parse_infos` are undefined in the context of `ClientOptions`.

**Comment by nicosomb** [comment]
Any news on this patch? When will it be released?

**Comment by farialima** [comment]
Hi @mnaberez, my last commit should cover your review, so this is ready for re-review and hopefully for merging...

**Comment by farialima** [comment]
@mnaberez, in 8b6b45f I've committed changes that, I believe, address your change request. Could you please approve my commit, and possibly merge the PR ?

**Comment by Gollam** [comment]
@mnaberez please respond, quite a few of us are waiting for this.

File: supervisor/options.py (hunk 1 above)
Comment (farialima) [reply supervisor/options.py]: Marking this as resolved, the commit above fixes it, I believe

**Comment by farialima** [comment]
I believe that this can be merged, it has been ready for close to a year, and I think it would help many people who (like me :) ) have spent a lot of time trying to understand what's wrong... @mnaberez please tell me if there's anything that would prevent it from being merged ?

**Comment by mnaberez** [comment]
> I believe that this can be merged

When I reviewed this patch initially, I tested it by actually trying to use it and it crashed because of undefined variables.  I see that specific error has been fixed now (thanks!).  I am open to reviewing it again.  Have you actually run Supervisor with this code and seen the new feature working?

**Comment by farialima** [comment]
Hi - I just saw your last comment. Yes I had run it in a a "real" supervisor, and seen the feature working... it's probably still worth merging :)

**Comment by mnaberez** [comment]
> Hi - I just saw your last comment. Yes I had run it in a a "real" supervisor, and seen the feature working... it's probably still worth merging :)

I'll give it another look.  Thanks!

**Comment by singhhrpreet** [comment]
@mnaberez Any news on this?

**Comment by farialima** [comment]
Hi @mnaberez, I believe this is ready for merging (and has been for some time :D ). Can you review it ? If there's anything you'd like me to add, I can look at it.

**Pull request merged**